### PR TITLE
3.10 compatibility

### PR DESCRIPTION
**Pull request by ppdebreuck**
Python 3.10 compatibility: random.choices weights must be finite. Used in FitGenetic. Fixed by replacing weights that are nans to 1e-5.

### CODE REVIEW

**Comment by ml-evs** [comment]
This seems to include all of the changes from #194, is that intentional?

**Comment by ppdebreuck** [comment]
Oops ! Fixed !

File: modnet/hyper_opt/fit_genetic.py [hunk 1]
```diff
@@ -96,9 +96,11 @@ def crossover(self, partner: Individual) -> Individual:
...
 
         child_genes = {
-            list(self.genes.keys())[i]: list(self.genes.values())[i]
-            if i in genes_from_mother
-            else list(partner.genes.values())[i]
+            list(self.genes.keys())[i]: (
+                list(self.genes.values())[i]
+                if i in genes_from_mother
+                else list(partner.genes.values())[i]
+            )
```
Comment (ml-evs) [inline modnet/hyper_opt/fit_genetic.py:103]: As it stands this is going to remake the genes lists for every iteration of the dict comprehension, could we tidy it up a bit?

(I know this change is just from auto-formatting)

something like

```python

genes_from_mother = random.sample(sorted(self.genes)), k=len(self.genes) // 2)
child_genes = {key: gene for key in self.genes if key in genes_from_mother else partner.genes[key]}
```

should work and just directly give you the keys to use?

File: modnet/hyper_opt/fit_genetic.py [hunk 2]
```diff
@@ -574,8 +576,10 @@ def run(
             weights = [
                 1 / lw**5 for lw in val_loss[ranking]
             ]  # **5 in order to give relatively more importance to the best individuals
+            weights = [1e-5 if math.isnan(weight) else weight for weight in weights]
             weights = [w / sum(weights) for w in weights]
             # selection: weighted choice of the parents -> parents with a low MAE have more chance to be selected
+            print(weights)
```
Comment (ml-evs) [inline modnet/hyper_opt/fit_genetic.py]:
```suggestion
```

File: modnet/hyper_opt/fit_genetic.py (hunk 1 above)
Comment (ppdebreuck) [reply modnet/hyper_opt/fit_genetic.py:103]: Indeed !

**Pull request merged**